analyzers/src/SonarAnalyzer.Common/Helpers/GeneratedCodeRecognizer.cs

```diff
@@ -62,12 +62,12 @@ public abstract class GeneratedCodeRecognizer
             tree is not null && (IsRazor(tree) || IsCshtml(tree));
 
         public static bool IsRazor(SyntaxTree tree) =>
-            tree.FilePath.EndsWith("razor.g.cs", StringComparison.OrdinalIgnoreCase)
-            || tree.FilePath.EndsWith("razor.ide.g.cs", StringComparison.OrdinalIgnoreCase);
+            // razor.ide.g.cs is razor generated files for design time builds. However, it is not considered here because of https://github.com/dotnet/razor/issues/9108
```

Comment:
```suggestion
            // razor.ide.g.cs is the extension for razor-generated files in the context of design-time builds. However, it is not considered here because of https://github.com/dotnet/razor/issues/9108
```